### ISSUE TITLE
Bug in UtcClock on linux?

### ISSUE THREAD

**Issue by martindederer** [pokowaka/jwt-cpp#3]
I was unable to validate tokens with proper "exp" and "iat" claims. Therefore i dug a little bit into the TimeValidator et al and stumbled upon what i think might be a bug.

```
class UtcClock : public IClock {
 public:
    uint64_t Now() {
      time_t rawtime = 0;
      struct tm ptm = {0};
      time(&rawtime);
      #ifdef _WIN32
      gmtime_s(&ptm, &rawtime);
      #else
      gmtime_r(&rawtime, &ptm);
      #endif
      return mktime(&ptm); // <----------------- mktime(...) should actually be timegm(...)?
    }
};
```

According to [the linux man pages](https://linux.die.net/man/3/gmtime_r) mktime takes broken down time expressed as local time and converts it to unix time. At the same time gmtime_r returns unix time. So if i am not mistaken mktime fails to return a correct time value when your local time is not UTC. The function "timegm" takes UTC and returns unix time.

Replacing

```
return mktime(&ptm);
```

with 

```
return timegm(&ptm);
```

should fix this issue.

**Comment by martindederer** [comment]
I wrote a small fix. I was not able to test it on windows.

See https://github.com/pokowaka/jwt-cpp/pull/4

**Comment by martindederer** [comment]
Resolved in commit eb9f3b5afc43f679c6bdc44196e6d3fc59ac6396.
